dbms/src/Storages/Transaction/PartitionStreams.cpp

```diff
@@ -167,7 +170,14 @@ static void writeRegionDataToStorage(
     /// Try read then write once.
     {
         if (atomic_read_write(false))
+        {
+            LOG_FMT_INFO(log, "[for test only] first decode sucess");
+
+            fiu_do_on(FailPoints::force_pause_query_until_write_finish, {
```

Comment:
Actually the role of this failpoint (what it does when it is enabled) is to unblock a query. How about naming it as: `FailPoints::unblock_query_init_after_write`?